### DISCUSSION TITLE
Loading Testing auth endpoint performance

### DISCUSSION THREAD

**Discussion by bmingles** [pocketbase/pocketbase#3959] [Q&A] [answered]
I've been load testing Pocketbase endpoint to get a feel for capacity, resource consumption, etc.
Test 1
- Get a token from auth-with-password endpoint
- Use the token to run 15k requests in batches of 100 against a collection endpoint that returns 2 small records.
This works great. Completes in ~ 30 seconds

Test 2
- Run 15k requests in batches of 100 against the auth-with-password endpoint
I let this one run for 5 minutes and it never finished

My questions:
1. Is it expected that hitting the auth endpoint would be significantly slower than a collection endpoint? If so, any idea by how much?
2. Since I never saw the test finish, I want to rule out a deadlock scenario. Is there anything with the auth endpoints that doesn't do well under load?

**Comment by ganigeorgiev** [comment, marked as answer]
For both questions the auth-with-password endpoint is expected to be slower due to the bcrypt hash verification.

By how much - it will depend on your running hardware. Locally is around 2r/s. For more elaborate results see for example the [results from the benchmarks repo](https://github.com/pocketbase/benchmarks/blob/master/results/hetzner_cax41.md#user-auth-with-password-expected-to-be-slow-due-to-passwordhash-verification).

**Reply by bmingles** [reply]
Is 2r/s 2 requests per second?

**Reply by ganigeorgiev** [reply]
Yes, but that's on my 12y old laptop and it would greatly depend on the number of concurrent requests you are performing. Please refer to the linked results from the benchmark repo for more details.

For Hetzner CAX11 (2vCPU ARM64, 4GB RAM) - https://github.com/pocketbase/benchmarks/blob/master/results/hetzner_cax11.md#user-auth-with-password-expected-to-be-slow-due-to-passwordhash-verification

For Hetzner CAX41 (16vCPU ARM64, 32GB RAM) - https://github.com/pocketbase/benchmarks/blob/master/results/hetzner_cax41.md#user-auth-with-password-expected-to-be-slow-due-to-passwordhash-verification
